asset-transfer-events/application-gateway-typescript/src/app.ts

```diff
@@ -0,0 +1,210 @@
+        identity: await newIdentity(),
+        signer: await newSigner(),
+        evaluateOptions: () => {
+            return { deadline: Date.now() + 5000 }; // 5 seconds
+        },
+        endorseOptions: () => {
+            return { deadline: Date.now() + 15000 }; // 15 seconds
+        },
+        submitOptions: () => {
+            return { deadline: Date.now() + 5000 }; // 5 seconds
+        },
+        commitStatusOptions: () => {
+            return { deadline: Date.now() + 60000 }; // 1 minute
+        },
+    });
+
+    let events: CloseableAsyncIterable<ChaincodeEvent> | undefined;
+
+    try {
+        // Get a network instance representing the channel where the smart contract is deployed.
+        const network = gateway.getNetwork(channelName);
+
+        // Get the smart contract from the network.
+        const contract = network.getContract(chaincodeName);
+
+        //Start Listening to events.
+        events = await startEventListening(network);
+
+        // Create a new asset on the ledger.
+        const firstBlockNumber = await createAsset(contract);
+
+        // Update an existing asset.
+        await updateAsset(contract)
+
+        // Transfer an existing asset.
+        await transferAsset(contract);
+
+        // Delete asset by assetID.
+        await deleteAssetByID(contract);
+
+        // Replay all the events received.
+        await Promise.race([timeout(10000),replayChaincodeEvents(network,firstBlockNumber)])
+
+    } finally {
+        events?.close();
+        gateway.close();
+        client.close();
+    }
+}
+
+main().catch(error => console.error('******** FAILED to run the application:', error));
+
+
+async function startEventListening(network: Network) {
+    console.log('\n*** Start chaincode event listening\n');
+    const events = await network.getChaincodeEvents(chaincodeName);
+    readEvents(events);
+    return events;
+}
+
+async function readEvents(events: CloseableAsyncIterable<ChaincodeEvent>): Promise<void> {
+    try {
+        for await (const event of events) {
+            const payload = utf8Decoder.decode(event.payload);
+            console.log(`\n<-- Chaincode event received, name: ${event.eventName}, payload: ${payload}, txID: ${event.transactionId}, blockNumber:${event.blockNumber}`);
+        }
+    } catch (error: unknown) {
+        if (!(error instanceof GatewayError) || error.code !== grpc.status.CANCELLED) {
+            throw error;
+        }
+    }
+}
+
+/**
+ * Submit a transaction asynchronously to create a new asset.
+ */
+async function createAsset(contract: Contract): Promise<bigint> {
+    console.log(`\n --> Submit Transaction: CreateAsset, creates ${assetId} owned by Tom with appraised value 100`);
+
+    const result = await contract.submitAsync(
+        'CreateAsset',
+        {
+            arguments: [
+                assetId,
+                'yellow',
+                '5',
+                'Tom',
+                '100',]
+        });
```

Comment:
For many of the other submit/evaluate calls you've left the transaction name on the first line and put arguments on subsequent lines. Whatever style you prefer is fine, it just might be nice to consistently use the same style throughout.